docs/tools/testing/testing_greentea.md

```diff
@@ -67,58 +71,190 @@ The build system looks for an `mbed_app.json` file in your shared project files
 
-The testing process requires that tests be built and that a test specification JSON file that describes these available tests exists. See the [test specification format](https://github.com/ARMmbed/mbed-os-tools/tree/master/packages/mbed-greentea#test-specification-json-formatted-input).
+Let's write your first test. Use Mbed CLI to create a new project:
+
+```
+$ mbed new first-greentea-test
+```
 
-The Greentea tool handles the actual testing process. To read more about this tool, please visit its [GitHub repository](https://github.com/ARMmbed/mbed-os-tools/tree/master/packages/mbed-greentea).
+As specified above, there is a convention where all tests live in the `TESTS/` directory. In the `first-greentea-test` folder create a folder `TESTS/tests/simple-test/`.
 
-#### Writing tests
+![Tree structure for Greentea tests](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/test01.png)
 
-You can write tests for your own project or add more tests to Mbed OS. You can write tests by using the [Greentea client](https://github.com/ARMmbed/mbed-os/tree/master/features/frameworks/greentea-client), and the [UNITY](https://github.com/ARMmbed/mbed-os/tree/master/features/frameworks/unity) and [utest](https://github.com/ARMmbed/mbed-os/tree/master/features/frameworks/utest) frameworks, which are located in `/features/frameworks`. Below is an example test that uses all of these frameworks:
+In this folder, create a file `main.cpp`. In here you can use UNITY, utest and the Greentea Client to write your test:
 
-```c++ TODO
+```cpp
 #include "mbed.h"
+#include "utest/utest.h"
+#include "unity/unity.h"
 #include "greentea-client/test_env.h"
-#include "unity.h"
-#include "utest.h"
-#include "rtos.h"
 
 using namespace utest::v1;
 
-// A test that returns successfully is considered successful
-void test_success() {
-    TEST_ASSERT(true);
-}
+// this is how a test looks
+static control_t simple_test(const size_t call_count) {
+    /* test content here */
+    TEST_ASSERT_EQUAL(4, 2 * 2);
 
-// Tests that assert are considered failing
-void test_failure() {
-    TEST_ASSERT(false);
+    return CaseNext;
 }
 
-utest::v1::status_t test_setup(const size_t number_of_cases) {
-    // Setup Greentea using a reasonable timeout in seconds
-    GREENTEA_SETUP(40, "default_auto");
-    return verbose_test_setup_handler(number_of_cases);
+utest::v1::status_t greentea_setup(const size_t number_of_cases) {
+    // here we specify the timeout (60s) and the host runner (the name of our Python file)
+    GREENTEA_SETUP(1*60, "default_auto");
+
+    return greentea_test_setup_handler(number_of_cases);
 }
 
-// Test cases
+// list of all cases
```

Comment:
This is listing test cases which are contained in this file and there are multiple files. "all" sounds me a bit more than one file tests